core/src/data_sources/node.rs

```diff
@@ -1,19 +1,81 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+use crate::project::Project;
+
+#[derive(Debug, Clone, Serialize, PartialEq, Deserialize, Copy)]
 pub enum NodeType {
     Document,
     Table,
     Folder,
 }
 
-#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+pub trait ToNode {
```

Comment:
@fontanierh what do you think of that pattern?

Reply:
We should use `TryFrom` IMHO

Reply:
Or just `From` actually since this can't fail